bindings/java/src/main/com/apple/foundationdb/ReadTransaction.java

```diff
@@ -132,7 +132,8 @@
 	 *
 	 * @param selector the relative key location to resolve
 	 *
-	 * @return a {@code CompletableFuture} which will be set to an absolute database key
+	 * @return a {@code CompletableFuture} which will be set to an absolute database key. 
+	 *         If no key is found, then a byte array with length 0 will be returned.
```

Comment:
I think this is where the wrinkle with having an `Exact` key selector is. For existing key selectors, a selector not finding a key would either return the empty key `""` if it resolved to the beginning of the database or `\xff` if it resolved to the end. Resolving an exact key selector would be a little bit different.

Resolving an exact key selector is a bit of a weird operation, but I guess the best way to think about it would be that you are doing a key existence check. I think the most general way to do this that fits within the current signature of this function would be to return the passed in key if it exists and `null` if not. To implement this at the Java level, you could implement this a couple of ways --

1. Just read the key using `get`
2. Resolve a `firstGreaterOrEqual` or `lastLessOrEqual` key selector and then check if the result is the same as the key you passed in

You could also just disallow resolving an exact key selector.

Reply:
so you raise a good point about correct behavior or `KeySelector.exact()`(my reply to your other comment talks about that briefly), but it's probably worth pointing out that the non-nullity of the byte[] actually pre-dates the `KeySelector.exact()` behavior(it comes from how the native layer is creating the returned byte[]), I'm just adding some javadoc here to make that explicit.

Reply:
The existing behavior for other key selectors shouldn't change, and it differs from the newly documented behavior in that it can also return the end of the key-space (`\xff` or `\xff\xff`) depending on the direction of your selector.